src/main/java/de/retest/recheck/RecheckImpl.java

```diff
@@ -165,13 +163,18 @@ public void capTest() {
 		suite.addTest( currentTestResult );
 		final TestReplayResult finishedTestResult = currentTestResult;
 		currentTestResult = null;
-		final Set<LeafDifference> uniqueDifferences = finishedTestResult.getDifferences( filter );
+
+		final GlobalIgnoreApplier stepIgnoreApplier = RecheckIgnoreUtil.loadRecheckIgnore( currentStepDir );
+		final Filter stepFilter = new CompoundFilter( Arrays.asList( suiteFilter, stepIgnoreApplier ) );
```

Comment:
Wouldn't this need 3 filters?

1. Global `recheck.ignore`
2. Suite `recheck.ignore`
3. Step `recheck.ignore`

Reply:
That's a good point - the three filters are already there, but in an obfuscated way: The global one is hiding in the misleadingly named `suiteFilter`, which is already constructed as a compound filter from global and suite filters in the constructor.
I guess we could either rename it to `globalAndSuiteFilter` (or similar) for clarification, or we can keep both filters in the `RecheckImpl` and then have the three item list you had expected. Any preference?

Reply:
Oh my! Good thing you asked about a test: It made me recheck my manual HTML test example to see if I could extract something unit-like. Instead, after a few trials my palm met my forehead: The code has an elementary conceptual problem (which is obvious in hindsight, don't know how I missed it then):

Currently, it only loads one step filter in `RecheckImpl::capTest`, i.e., only for the last step in a test. In general, `RecheckImpl` does step-wise stuff _mostly_ via the `ActionReplayResult`s, but `capTest` can currently only use one suite+global filter when finalizing the test results.

Unfortunately, I see no elegant way to fix this: Both `TestReplayResult::getDifferences` and the whole `TestReplayResultPrinter` currently also assume a single, constant filter when accumulating results from different `ActionReplayResult`s. One alternative would be for them to use a `Map<ActionReplayResult, Filter>` to look up the (step-specific) filter per action; but this seems a little hackish to me, and it would change more public signatures and associated behaviour than this feature might be worth. I have checked these changes in so you can take a look yourself.

I have also verified the changed code with my little manual html test (which Github won't let me attach here, strangely), and I am very certain that it now works as desired, but I would understand completely if you think it might be better to close this pull request for the moment. (The suite filters, part 1 of this issue, are not affected.)